includes/breadcrumb.inc

```diff
@@ -24,10 +24,19 @@
  *   not including the given object. For use in the function
  *   drupal_set_breadcrumb().
  */
-function islandora_get_breadcrumbs($object) {
-  $breadcrumbs = array();
+function islandora_get_breadcrumbs(AbstractObject $object) {
   if (variable_get('islandora_render_drupal_breadcrumbs', TRUE)) {
-    $breadcrumbs = islandora_get_breadcrumbs_recursive($object->id, $object->repository);
+    $backend = variable_get('islandora_breadcrumbs_backends', ISLANDORA_BREADCRUMB_LEGACY_BACKEND);
+    $backends = module_invoke_all('islandora_breadcrumbs_backends');
+    if ($backend === ISLANDORA_BREADCRUMB_LEGACY_BACKEND || !isset($backends[$backend])) {
+      $breadcrumbs = islandora_get_breadcrumbs_recursive($object->id, $object->repository);
+    }
+    else {
+      if (isset($backends[$backend]['file'])) {
+        require_once $backends[$backend]['file'];
+      }
+      $breadcrumbs = call_user_func($backends[$backend]['callable'], $object);
+    }
     array_pop($breadcrumbs);
```

Comment:
As you mentioned in Jira, this array_pop() could be part of the default implementation and let each custom hook include it, or not.